src/service-worker/core.js

```diff
@@ -225,6 +244,109 @@ export function generateFallbackClientId(referrer) {
 }
 
 /**
+ * Fetches a URL, and stores it into the cache if the response is valid.
+ * Repeated fetches of the same URL will be batched into a single request while
+ * the first is still fetching.
+ *
+ * @param {!Cache} cache
+ * @param {!Request} request
+ * @return {!Promise<!Response>}
+ * @visibleForTesting
+ */
+export function fetchAndCache(cache, request) {
+  const url = request.url;
+
+  // Batch fetches. Mainly for the /diversions endpoint.
+  if (fetchPromises[url]) {
+    return fetchPromises[url];
+  }
+
+  return fetchPromises[url] = cache.match(request)
+      .then(response => {
+        if (response && !expired(response)) {
```

Comment:
This is only relevant for the diversions endpoint, right? Because rtv URLs practically never expire.